recipes/master.rb

```diff
@@ -44,33 +53,35 @@
   notifies :reload, 'ohai[salt]', :immediate
 end
 
-if Chef::Config[:solo]
-  log 'Salt key exchange not supported on Chef solo' do
-    level :warn
-  end
-else
-  minion_search = "role:#{node['salt']['role']['minion']}"
-  if node['salt']['master']['environment']
-    minion_search += " AND chef_environment:#{node['salt']['master']['environment']}"
-  end
+if node['salt']['key_accept_method'] == 'pub_key_sync'
```

Comment:
Lets move the code in this block to a new recipe called `master_key_accept` and only include it if `pub_key_sync`